### ISSUE TITLE
Unable to add an error, column should be a positive number but -243 given

### ISSUE THREAD

**Issue by rachel-carvalho** [AtomLinter/linter-jscs#86]
I was editing a JS file, moving a line to the top (`CMD` + `CTRL` + `up`) and got this error. I don't know if it's a linter-jscs or linter problem.

This error keeps being thrown every time I move a line and that causes a lint error on the same file. I changed tabs to check if another file threw it and it didn't. I closed and re-opened the tab and it still throws this error.

After quiting Atom and re-opening it, it still happens. It also happens if I cut the line I'm moving and paste it to an invalid line or if I write an invalid on the same place.

```
AssertionError: Unable to add an error, `column` should be a positive number but -243 given
  at Object.Errors._validateInput (/Users/rachelcarvalho/.atom/packages/linter-jscs/node_modules/jscs/lib/errors.js:73:9)
  at Object.Errors.add (/Users/rachelcarvalho/.atom/packages/linter-jscs/node_modules/jscs/lib/errors.js:46:14)
  at [object Object].StringChecker._addParseError (/Users/rachelcarvalho/.atom/packages/linter-jscs/node_modules/jscs/lib/string-checker.js:205:16)
  at [object Object].<anonymous> (/Users/rachelcarvalho/.atom/packages/linter-jscs/node_modules/jscs/lib/string-checker.js:97:22)
  at Array.forEach (native)
  at [object Object].StringChecker.checkString (/Users/rachelcarvalho/.atom/packages/linter-jscs/node_modules/jscs/lib/string-checker.js:95:31)
  at Object.lint (/Users/rachelcarvalho/.atom/packages/linter-jscs/index.js:124:36)
  at /Users/rachelcarvalho/.atom/packages/linter/lib/editor-linter.coffee:69:26
  at currentLinter (/Users/rachelcarvalho/.atom/packages/linter/lib/editor-linter.coffee:68:20)
  at /Users/rachelcarvalho/.atom/packages/linter/lib/editor-linter.coffee:82:23
  at Set.forEach (native)
  at EditorLinter.triggerLinters (/Users/rachelcarvalho/.atom/packages/linter/lib/editor-linter.coffee:64:26)
  at /Users/rachelcarvalho/.atom/packages/linter/lib/editor-linter.coffee:57:27
```

I was able to reproduce the error with the following JS, if you try to paste `barred = new Bar();` just after `fookey: 1,`:
```javascript
var Bar = require('bar');

module.exports = {
  foo: {
    fookey: 1,
    fookey2: 2
  },

  method: function() {
    barred = new Bar();
  }
};
```

**Comment by scholtzm** [comment]
Getting this error as well.

**Comment by catscarlet** [comment]
I got the similar error today.

```
AssertionError: Unable to add an error, column should be a positive number but -367 given
AssertionError: Unable to add an error, `column` should be a positive number but -367 given
  at Object.Errors._validateInput (C:\Users\sunfu\.atom\packages\linter-jscs\node_modules\jscs\lib\errors.js:73:9)
  at Object.Errors.add (C:\Users\sunfu\.atom\packages\linter-jscs\node_modules\jscs\lib\errors.js:46:14)
  at [object Object].StringChecker._addParseError (C:\Users\sunfu\.atom\packages\linter-jscs\node_modules\jscs\lib\string-checker.js:212:16)
  at [object Object].<anonymous> (C:\Users\sunfu\.atom\packages\linter-jscs\node_modules\jscs\lib\string-checker.js:97:22)
  at Array.forEach (native)
  at [object Object].StringChecker.checkString (C:\Users\sunfu\.atom\packages\linter-jscs\node_modules\jscs\lib\string-checker.js:95:31)
  at Object.lint (C:\Users\sunfu\.atom\packages\linter-jscs\index.js:83:36)
  at Promise.then._this.emitter.emit.linter (C:\Users\sunfu\.atom\packages\linter\lib\linter-registry.coffee:59:22)
  at LinterRegistry.triggerLinter (C:\Users\sunfu\.atom\packages\linter\lib\linter-registry.coffee:58:16)
  at C:\Users\sunfu\.atom\packages\linter\lib\linter-registry.coffee:52:17
  at Array.map (native)
  at C:\Users\sunfu\.atom\packages\linter\lib\linter-registry.coffee:50:27
```


This is what I was editing

```
var zNodes = [
  { id:1, pId:0, name:"Question1:which is apple", open=true, isParent:true, nocheck:true },
  { id:11, pId:1, name:"apple", purl:"apple.jpg" },
  { id:12, pId:1, name:"banana", purl:"banana.jpg" },
  { id:13, pId:1, name:"pear", purl:"pear.jpg" },
  { id:2, pId:0, name:"Question2:which is bike", open:true, isParent:true, nocheck:true },
  { id:21, pId:2, name:"car" },
  { id:22, pId:2, name:"bus" },
  { id:23, pId:2, name:"bike" }
];
```

I know there is a mistake at line 2:
```
open=true  //should be "open:true"
```

but linter-jscs give out a AssertionError instead of show syntax error of my code

![snap030](https://cloud.githubusercontent.com/assets/10594035/10387911/cca6bb2a-6e97-11e5-9d7d-32b9cd3ec9b2.png)

![snap028](https://cloud.githubusercontent.com/assets/10594035/10387893/ae7e0e82-6e97-11e5-883a-bc2e528e62dc.png)

**Comment by catscarlet** [comment]
my atom version is 1.0.19
jscs-fixer version is 1.0.2
linter version is 1.8.1
linter-jscs version is 3.1.1
system win7x64

.jscsrc
```
{
    "preset": "jquery",
    "requireCapitalizedComments": false,
    "disallowMixedSpacesAndTabs": true,
    "requireCamelCaseOrUpperCaseIdentifiers": "ignoreProperties",
    "maximumLineLength": false,
    "requireDotNotation": false,
    "requireSpacesInsideParentheses": false,
    "validateQuoteMarks": false,
    "validateIndentation": 2
}

```

sorry for missing these information of the last comment.

**Comment by rdebeasi** [comment]
I'm experiencing the same issue. In the code below, I typed `=` instead of `:`. The error message disappeared when I fixed the mistake in my code.

```js
CardSwiper.prototype =
	{
		busy: false,
		count: 0,
		on: 1,
		size: 0,
		$imagesToLoad: $(),
		needsToPreload = false, // Whoops!!
	}
```

**Comment by Arcanemagus** [comment]
When you see that error if you save the file and then run `jscs` from the command line does it also have the same issue? This looks like a problem in `jscs` rather than this plugin.

**Comment by rdebeasi** [comment]
You're right - I do get the same error when running `jscs` from the command line.

**Comment by Arcanemagus** [comment]
Marking this as closed then as it is a `jscs` bug.

They will probably appreciate a bug report over there, especially if you have a file that is able to reproduce this for them :wink:

It looks like this might have been reported before here https://github.com/jscs-dev/node-jscs/issues/1006, but that was _supposedly_ fixed in `jcsc` v1.11, so you might want to file a new issue referencing the older one.
